### PR TITLE
Fixing the by region update transfer of the compressed texture

### PR DESCRIPTION
**Pull request by samcake**
During the Texture Streaming, for large images (more than 4mb) we chunk their transfer from sys mem to gpu mem  in regions.
For compressed format, these regions of update (using glCompressedTexSubImage2D) MUST be a multiple of 4x4 to respect the underneath tiling of the compressed images.
WE weren't doing that, this pr fixes that.

## TEST PLAN
Add this model to your domain: https://hifi-public.s3.amazonaws.com/huffman/ktx/builder-grid/builder-grid.baked.fbx
In Current dev master: you see that the floor resolution improves until it turns black (that s the top mip not beeing transfered properly and so the texture contains garbage for higher resolution mip).

With this PR, this effect doesn't happen.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84305c20498fe5ceda8fea22b214770c3b5f83cc available at: https://gist.github.com/9bac8dc47a1ba62fa6eb82724e22d30b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-84305c2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-84305c2.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c583ffbac447d865914442817eaa5323b1fc67fd available at: https://gist.github.com/9bac8dc47a1ba62fa6eb82724e22d30b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-c583ffb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-c583ffb.dmg

**Comment by samcake** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: c583ffbac447d865914442817eaa5323b1fc67fd available at: https://gist.github.com/9bac8dc47a1ba62fa6eb82724e22d30b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-c583ffb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-c583ffb.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c583ffbac447d865914442817eaa5323b1fc67fd available at: https://gist.github.com/9bac8dc47a1ba62fa6eb82724e22d30b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-c583ffb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-c583ffb.dmg

File: libraries/gpu-gl/src/gpu/gl45/GL45BackendVariableTexture.cpp [hunk 1]
```diff
@@ -196,11 +196,14 @@ void GL45ResourceTexture::populateTransferQueue() {
 
             // break down the transfers into chunks so that no single transfer is 
             // consuming more than X bandwidth
+            // For compressed format, regions must be a multiple of the 4x4 tiles, so enforce 4 lines as the minimum block
             auto mipSize = _gpuObject.getStoredMipFaceSize(sourceMip, face);
             const auto lines = mipDimensions.y;
-            auto bytesPerLine = mipSize / lines;
+            const uint32_t BLOCK_NUM_LINES { 4 };
```
Comment (jherico) [inline libraries/gpu-gl/src/gpu/gl45/GL45BackendVariableTexture.cpp]: what happens when the mip sizes get below 4x4?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26441dbf2feb6ae49125f4b609d0592fd8b55106 available at: https://gist.github.com/9bac8dc47a1ba62fa6eb82724e22d30b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-26441db.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-26441db.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e2e6b1f9f185d991f8aae00329af688d1950b286 available at: https://gist.github.com/9bac8dc47a1ba62fa6eb82724e22d30b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-e2e6b1f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10359-e2e6b1f.dmg

**Comment by Nex-Pro** [comment]
Testing this one.

**Comment by Nex-Pro** [comment]
@samcake 

Problem is absent in this build but also I didn't see the problem in 6442.

**Pull request merged**